client/src/main/java/io/split/engine/common/SyncManagerImp.java

```diff
@@ -74,6 +81,9 @@ public class SyncManagerImp implements SyncManager {
         _telemetrySynchronizer = checkNotNull(telemetrySynchronizer);
         _config = checkNotNull(config);
         _startingSyncCallBackoffBaseMs = config.startingSyncCallBackoffBaseMs();
+        _impressionManager = impressionsManager;
+        _eventsTask = eventsTask;
+        _telemetrySyncTask = telemetrySyncTask;
```

Comment:
If these are not nullable, maybe `checkNotNull` could be added?
```suggestion
        _impressionManager = checkNotNull(impressionsManager);
        _eventsTask = checkNotNull(eventsTask);
        _telemetrySyncTask = checkNotNull(telemetrySyncTask);
```

Reply:
Now I'm checking.